src/nix.rs

```diff
@@ -0,0 +1,338 @@
+use crate::cache::Cache;
+use crate::parser::utils::mk_span;
+use crate::term::{BinaryOp, UnaryOp};
+use crate::term::{MergePriority, MetaValue, RichTerm, Term};
+use codespan::FileId;
+use rnix;
+use rnix::types::{BinOp, EntryHolder, TokenWrapper, TypedNode, UnaryOp as UniOp};
+use std::collections::HashMap;
+
+impl From<(UniOp, FileId)> for RichTerm {
```

Comment:
It sounds a bit overstretched to implement the from trait from a tuple of an unary op and a file id into a `RichTerm`. Usually `From/Into` are for somehow straightforward conversions between. E.g. in Rust I think you can't do `let s : String = 5.into()`, because it implies a non trivial operation. You need to do `5.to_string()`, where `to_string()` has exactly the same signature as `into()`, but has a different name. What about implementing something like `translate_nix_op` or whatever rather than the `From` trait?